client-ts/Microsoft.AspNetCore.SignalR.Client.TS/HubConnection.ts

```diff
@@ -125,17 +127,42 @@ export class HubConnection {
         return this.connection.stop();
     }
 
-    invoke(methodName: string, ...args: any[]): Promise<any> {
-        let id = this.id;
-        this.id++;
+    stream<T>(methodName: string, ...args: any[]): Observable<T> {
+        let invocationDescriptor = this.createInvocation(methodName, args);
 
-        let invocationDescriptor: InvocationMessage = {
-            type: MessageType.Invocation,
-            invocationId: id.toString(),
-            target: methodName,
-            arguments: args,
-            nonblocking: false
-        };
+        let subject = new Subject<T>();
+
+        this.callbacks.set(invocationDescriptor.invocationId, (invocationEvent: CompletionMessage | ResultMessage) => {
+            if (invocationEvent.type === MessageType.Completion) {
+                let completionMessage = <CompletionMessage>invocationEvent;
+                if (completionMessage.error) {
+                    subject.error(new Error(completionMessage.error));
+                }
+                else if(completionMessage.result) {
+                    subject.error(new Error("Server provided a result in a completion response to a streamed invocation."));
+                }
+                else {
+                    // TODO: Log a warning if there's a payload?
+                    subject.complete();
+                }
+            }
+            else {
+                subject.next(<T>(<ResultMessage>invocationEvent).item);
+            }
+        });
+
+        //TODO: separate conversion to enable different data formats
```

Comment:
It's not a new comment actually: https://github.com/aspnet/SignalR/blob/dev/client-ts/Microsoft.AspNetCore.SignalR.Client.TS/HubConnection.ts#L156

@moozzyk was an issue filed for this at the time it was added?